src/idl/include/idl/tree.h

```diff
@@ -118,6 +120,8 @@ enum idl_type {
   IDL_LDOUBLE = (IDL_BASE_TYPE | IDL_FLOATING_PT_TYPE | 3u)
 };
 
+#define IDL_TYPE_MASK ((IDL_TYPEDEF << 1) - 1)
+
```

Comment:
`idl_type` takes care of this. Not sure if that covers your usecase, but my idea was that the backend doesn't have to check masks but use `idl_type`, `idl_is_integer_type`, etc.

Reply:
This define is only used in `idl_type`, which before this change had a line: 

`mask = idl_mask(node) & ((IDL_TYPEDEF << 1) - 1);`

But as @eboasson suggested in his review, the definition of the max relevant enum value (which currently is `IDL_TYPEDEF`, but may change) should be close to its definition, so therefore I defined `IDL_TYPE_MASK` there.